sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1583,6 +1611,35 @@ class Analyzer(
           failAnalysis(s"Invalid usage of '*' in expression '${o.prettyName}'")
       }
     }
+
+    // For aggregateExpression we need NamedExpression, so for top level Struct field
+    // Alias, we won't trim it, for non top level Struct field Alias, we need to trim
+    // it since we have trimmed groupByExpressions.
```

Comment:
the comment relies on `trimStructFieldAlias`, can we move this method after that?

Reply:
> the comment relies on `trimStructFieldAlias`, can we move this method after that?

Done